app/controlplane/internal/biz/attestationstate.go

```diff
@@ -0,0 +1,236 @@
+		return nil, fmt.Errorf("failed to unmarshal attestation state: %w", err)
+	}
+
+	return &AttestationState{State: state}, nil
+}
+
+func (uc *AttestationStateUseCase) Reset(ctx context.Context, workflowID, runID string) error {
+	run, err := uc.checkWorkflowRunInWorkflow(ctx, workflowID, runID)
+	if err != nil {
+		return fmt.Errorf("failed to check workflow run: %w", err)
+	}
+
+	if err := uc.repo.Reset(ctx, run.ID); err != nil {
+		return fmt.Errorf("failed to reset attestation state: %w", err)
+	}
+
+	return nil
+}
+
+// checkWorkflowRunInWorkflow checks if the workflow run belongs to the provided workflow
+// This is important because the workflow is something that comes embedded in the auth token
+// so it can be used to make sure the user is not spoofing a different run that doesn't have access to
+func (uc *AttestationStateUseCase) checkWorkflowRunInWorkflow(ctx context.Context, workflowID, runID string) (*WorkflowRun, error) {
+	workflowUUID, err := uuid.Parse(workflowID)
+	if err != nil {
+		return nil, NewErrInvalidUUID(err)
+	}
+
+	runUUID, err := uuid.Parse(runID)
+	if err != nil {
+		return nil, NewErrInvalidUUID(err)
+	}
+
+	run, err := uc.wfRunRepo.FindByID(ctx, runUUID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to workflow run: %w", err)
+	} else if run == nil {
+		return nil, NewErrNotFound("workflow run")
+	}
+
+	if run.Workflow.ID != workflowUUID {
+		return nil, NewErrNotFound("workflow run")
+	}
+
+	return run, nil
+}
+
+// The following code is in charge of symmetric encryption and decryption of the attestation state
+// The only purpose is to have encryption at rest in the database
+const (
+	saltSize   = 16
+	iterations = 10000
+	keySize    = 32 // AES-256
+	// The magic string is used to check if the passphrase is correct
+	// It's prepended to the plaintext before encryption
+	// If the passphrase is incorrect, the decrypted data won't start with this string
+	magic = "Salted__"
+)
+
+// Generate an AES key derived from the passphrase and a salt
+func generateKey(passphrase string, salt []byte) []byte {
+	return pbkdf2.Key([]byte(passphrase), salt, iterations, keySize, sha256.New)
+}
+
+func encrypt(data []byte, passphrase string) ([]byte, error) {
+	salt := make([]byte, saltSize)
+	if _, err := io.ReadFull(rand.Reader, salt); err != nil {
+		return nil, err
+	}
+
+	key := generateKey(passphrase, salt)
+
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+	// Prepend magic string to the plaintext
+	plaintextWithMagic := append([]byte(magic), data...)
+
+	// 6MB limit to protect against allocation overflows
+	if len(plaintextWithMagic) > 6*1024*1024 {
+		return nil, errors.New("value too large")
+	}
+
+	ciphertext := make([]byte, aes.BlockSize+len(plaintextWithMagic))
+	iv := ciphertext[:aes.BlockSize]
+	if _, err := io.ReadFull(rand.Reader, iv); err != nil {
+		return nil, err
+	}
+
+	stream := cipher.NewCFBEncrypter(block, iv)
```

Comment:
I would document the reasons for choosing CFB and whether any best practices, patterns, or examples for symmetric encryption were followed here.